examples/text_analysis_openai_gpt.py

```diff
@@ -0,0 +1,152 @@
+# Databricks notebook source
+# MAGIC %md
+# MAGIC # Text analysis with DiscoverX & Azure OpenAI
+# MAGIC
+# MAGIC This notebooks uses [DiscoverX](https://github.com/databrickslabs/discoverx) to analyze text with [AZURE OpenAI API](https://learn.microsoft.com/en-us/azure/ai-services/openai/chatgpt-quickstart?tabs=command-line&pivots=programming-language-studio) over a set of tables in Unity Catalog.
+# MAGIC
+# MAGIC The notebook will:
+# MAGIC 1. Use DiscoverX to sample a set of tables from Unity Catalog and unpivot all string columns into a long format dataset
+# MAGIC 2. Run text analysis with Azure OpenAI GPT model
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ## Install dependencies
+
+# COMMAND ----------
+
+# MAGIC %pip install openai
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ## Setup widgets
+
+# COMMAND ----------
+
+dbutils.widgets.text("secret_scope", "discoverx", "Secret Scope")
+dbutils.widgets.text("open_ai_base", "openaibase", "Secret Key of Open API Base")
+dbutils.widgets.text("open_ai_key", "openaikey", "Secret Key of Open AI API Key")
+dbutils.widgets.text("from_tables", "discoverx_sample.*.*", "from tables")
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ## Import required libs and initialize variables
+
+# COMMAND ----------
+
+import openai
+import pandas as pd
+from pyspark.sql.functions import (
+    pandas_udf,
+    col,
+    concat,
+    lit,
+    explode,
+    count,
+    avg,
+    min,
+    max,
+    sum,
+    collect_set,
+    concat_ws,
+    regexp_replace,
+    upper,
+)
+from pyspark.sql.types import ArrayType, StringType, StructType, FloatType, StructField
+from typing import Iterator
+
+# COMMAND ----------
+
+from_tables = dbutils.widgets.get("from_tables")
+
+# Set the sample rows size
+sample_size = 100
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ## Initialize discoverx
+
+# COMMAND ----------
+
+from discoverx import DX
+
+dx = DX()
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ## Transform all sampled tables
+
+# COMMAND ----------
+
+unpivoted_df = (
+    dx.from_tables(from_tables)
+    .unpivot_string_columns(sample_size=sample_size)
+    .to_union_dataframe()
+    .localCheckpoint()  # Checkpointing to reduce the query plan size
+)
+
+# unpivoted_df.display()
+
+# COMMAND ----------
+
+display(unpivoted_df)
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ### Define udf to use gpt apis
+
+# COMMAND ----------
+
+openai_base_broadcast = sc.broadcast(
+    dbutils.secrets.get(dbutils.widgets.get("secret_scope"), dbutils.widgets.get("open_ai_base"))
+)
+openai_key_broadcast = sc.broadcast(
+    dbutils.secrets.get(dbutils.widgets.get("secret_scope"), dbutils.widgets.get("open_ai_key"))
+)
+
+# COMMAND ----------
+
+# Define the UDF function
+@pandas_udf(StringType())
+def predict_value_udf(s):
```

Comment:
why don't we use the ai gateway in this example? Should be possible correct?
Same question about costs as above ...

Reply:
Made changes to use gateway for open ai